spyder/app/tests/test_mainwindow.py

```diff
@@ -3470,31 +3468,24 @@ def test_path_manager_updates_clients(qtbot, main_window, tmpdir):
 
     cmd = 'import sys;print(sys.path)'
 
-    # Check Spyder is updated
-    main_window.console.execute_lines(cmd)
-    syspath = main_window.console.get_sys_path()
-    assert folder in syspath
+    # Check that there is at least one shell
+    shells = [c.shellwidget for c in main_window.ipyconsole.get_clients()
+              if c is not None]
+    assert len(shells) >= 1
 
     # Check clients are updated
-    count = 0
-    for client in main_window.ipyconsole.get_clients():
-        shell = client.shellwidget
-        if shell is not None:
-            control = shell._control
-            control.setFocus()
-
-            qtbot.waitUntil(lambda: shell._prompt_html is not None,
-                            timeout=SHELL_TIMEOUT)
-
-            with qtbot.waitSignal(shell.executed):
-                syspath = shell.execute(cmd)
-
-            # `shell.executed` signal was not working so we use waitUntil
-            qtbot.waitUntil(lambda: test_folder in control.toPlainText(),
-                            timeout=SHELL_TIMEOUT)
-            assert test_folder in control.toPlainText()
```

Comment:
@goanpeca @dalthviz , why should `qtbot` wait for `test_folder in control.toPlainText()` to be True? This is the assertion condition, so if the shell failed to update, `qtbot` will unnecessarily wait for 20-30 seconds before moving forward to the assertion statement. If the intent is to ensure that the text is written to the console before checking the assertion, then isn't this satisfied by the previous `qtbot.waitSignal(shell.executed)`? Perhaps a `SHELL_TIMEOUT` added here as well?